src/lexer/lex.c

```diff
@@ -24,3 +24,110 @@ int unlex(Lexer *l, Token *t) {
     l->unlexed_count++;
     return 0;
 }
+
+TokenType ttype_one_char(char c) {
+    switch (c) {
+    case '(':
+        return TT_OPAREN; // (
+    case ')':
+        return TT_CPAREN; // )
+    case '{':
+        return TT_OBRACE; // {
+    case '}':
+        return TT_CBRACE; // }
+    case '[':
+        return TT_OBRACKET; // [
+    case ']':
+        return TT_CBRACKET; // ]
+    case ';':
+        return TT_SEMI; // ;
+    }
+
+    return TT_NO_TOKEN;
+}
+
+TokenType ttype_many_chars(const char *contents) {
+    // Includes only numbers
+    int all_numeric = 1;
+
+    int count_fs = 0;
+    int count_us = 0;
+
+    int i;
+
+    if (contents == NULL) {
+        return TT_NO_TOKEN;
+    }
+
+    // Does it start with a negative sign
+    if (contents[0] != '\0') {
```

Comment:
The logic here is redundant methinks (you only need the inner `if`), also why would we return an identifier? Wouldn't this return `TT_IDENTIFIER` for the string `-=` which is an operator?

Reply:
Yup, removed outer if

Reply:
Will the lexer lex `-=`  as a single token or will it do `-` and `=` separately?